src/librustc/mir/mod.rs

```diff
@@ -1660,18 +1658,15 @@ impl<'tcx> TerminatorKind<'tcx> {
                 values
                     .iter()
                     .map(|&u| {
-                        let mut s = String::new();
-                        let c = ty::Const {
+                        (&ty::Const {
```

Comment:
You can use `tcx.mk_const(...)`, and even create a `FmtPrinter` yourself if you want to avoid some unnecessary lifting.

But IMO `fmt_successor_labels` should return an iterator of something that implements `ty::Print`, or at least `fmt::Display`.
The entirety of MIR formatting could use the `ty::print{,::pretty}` API (feel free to open an issue about that)

Reply:
But that will intern a bunch of constants that we never need again. The same goes for match checking which also uses `ty::Const` directly. I'm not worried about the unnecessary lifting, I'm worried about the unnecessary interning.

Reply:
They're small integers so I'm not worried - also, we only use this when dumping MIR, so there isn't really much happening afterwards to be affected by memory pressure.